### PR TITLE
bug fix for determining AuthMethod correctly

### PR DESCRIPTION
**Pull request by pdowler**

### CODE REVIEW

File: cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java [hunk 1]
```diff
@@ -1104,37 +1104,63 @@ protected RegistryClient getRegistryClient()
...
+                boolean domainMatch = false;
+                String domain = NetUtil.getDomainName(serviceURL);
+                for (SSOCookieCredential cc : subject.getPublicCredentials(SSOCookieCredential.class))
+                {
+                    if (cc.getDomain().equals(domain))
+                        domainMatch = true;
+                } 
+                if (!domainMatch)
+                {
+                    throw new AccessControlException("No valid public credentials.");
```
Comment (brianmajor) [inline cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java]: A more explicit error message:  No valid public credentials for this domain.

File: cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java [hunk 2]
```diff
@@ -1104,37 +1104,63 @@ protected RegistryClient getRegistryClient()
...
+    
+    private AuthMethod getAuthMethod(Subject subject)
     {
-        Subject subject = AuthenticationUtil.getCurrentSubject();
         if (subject != null)
         {
-            for (Object o : subject.getPublicCredentials())
+            // web services use CDP to load a proxy cert so prefer that
+            X509CertificateChain privateKeyChain = X509CertificateChain.findPrivateKeyChain(
+                    subject.getPublicCredentials());
```
Comment (brianmajor) [inline cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java]: Indent the three lines above

File: cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java (hunk 1 above)
Comment (pdowler) [inline cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java]: message improved

File: cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java (hunk 2 above)
Comment (pdowler) [inline cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java]: the loop is gone so it looks correctly indented to me

Comment (brianmajor) [inline cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java]: Sorry, you're right, I misread the diff

**Pull request merged**